### DISCUSSION TITLE
When can we have a support for webhooks?

### DISCUSSION THREAD

**Discussion by rakesh-groww** [microcks/microcks#915] [Q&A]
I'm mocking a third party service. This third party sends me a webhook on certain events like change of status. The mocked service should also be able to send me a webhook against a pervious sync api.

**Comment by lbroudoux** [comment]
Hello,

Unfortunately, webhooks are not supported by Microcks at the moment, but we've targeted them on our roadmap (see https://github.com/orgs/microcks/projects/1?pane=issue&itemId=26464135). We were waiting for someone who requested it 😉 

Would you mind helping us with a clear explanation on what you need, how you expect Microcks to handle that, a quick demonstration scenario,.... whatever could help us figure out how to implement that the best way.

Thanks a lot!

**Reply by AniketGroww** [reply]
For example,
So, suppose there is an api which creates an order, now, as soon as an user places an order we show, "order initiated" which is the status of the order at that time that we have received from 3rd party service.

It looks like this:-

create api:-
user -> our service <-> 3rd party service (which we want to mock) for testing.    (ORDER INITIATED FLOW)

But in the prod env, after this, when suppose that order is verified, the 3rd party sends a webhook to our service which is supposed to change the status of the orderId to INITIATED to SUCCESS.

Now, in the current scenario of microcks, we are not able to implement this.

@lbroudoux i hope this example kindoff clarifies the requirement

**Reply by lbroudoux** [reply]
This is a great start! Thank you! 
Do you think you could translate that into samples OpenAPI operations spec we could use as a reference for later implementing the feature? 
I also remind me a conversation I started previously that may be relevant in that context: https://github.com/orgs/microcks/discussions/694 Would you mind sharing your idea on that?

**Reply by AniketGroww** [reply]
Yeah cool, no issues. If possible, we would also love to contribute to the same.

**Reply by AniketGroww** [reply]
https://github.com/orgs/microcks/discussions/694#discussioncomment-7048840
Have added my thoughts ! @lbroudoux

**Reply by lbroudoux** [reply]
Will read 'em thoroughly!

**Comment by rakesh-groww** [comment]
Can we use openapi webhook documentation for this requirement
https://bump.sh/blog/documenting-your-OpenAPI-webhooks
https://github.com/OAI/OpenAPI-Specification/blob/main/examples/v3.1/webhook-example.yaml

**Reply by lbroudoux** [reply]
Hello! 
I spent some time thinking and writing about adding support to OpenAPI callbacks, OpenAPI webhooks and Sync-to-Async trigger in Microcks. I wrote everything as a GitHub discussion issue here: https://github.com/orgs/microcks/discussions/1039
Please take some time to read this and give your feedback.
